File: ios/AirSnapReactNative/AirSnapReactNative.m
//
//  AirSnapReactNative.m
//  AirSnapReactNative
//
//  Created by Richard Lam on 7/24/21.
//

#import <React/RCTBridgeModule.h>
#import <AirSnapSDK/AirSnapSDK-Swift.h>

@interface RCT_EXTERN_MODULE(AirSnapReactNative, NSObject)

RCT_EXTERN_METHOD(start:(NSString *)withAppKey privacyRule:(NSString *)privacyRule)

RCT_EXPORT_METHOD(getSessionId:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject) {
    resolve([AirSnap sessionId]);
}

RCT_EXPORT_METHOD(getUserId:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject) {
    resolve([AirSnap userId]);
}

RCT_EXTERN_METHOD(setUserId:(NSString *)id)

RCT_EXTERN_METHOD(captureException
                  :(NSDictionary *)exception
                  level:(NSString *)level
                  context:(NSDictionary *)context)

@end

File: ios/AirSnapReactNative/AirSnapReactNative.swift
//
//  AirSnapReactNative.swift
//  AirSnapReactNative
//
//  Created by Richard Lam on 7/24/21.
//

import Foundation
import AirSnapSDK

@objc(AirSnapReactNative)
class AirSnapReactNative: NSObject {

    @objc(requiresMainQueueSetup)
    static func requiresMainQueueSetup() -> Bool {
        return false
    }
    
    @objc func start(_ withAppKey: String, privacyRule: String) -> Void {
        AirSnap.start(withAppKey: withAppKey, privacyRule: privacyRule == "unmask" ? .unmask : .mask)
    }

    @objc func setUserId(_ id: String?) -> Void {
        AirSnap.setUser(id: id)
    }
    
    @objc public func captureException(_ issue: [String: NSObject], level: String, context: [String: NSObject]) {
      let issueInput = IssueInput(platform: issue["platform"] as! String , summary: issue["summary"] as! String, data: issue["data"] as! [String: NSObject]?)

      AirSnap.captureException(issue: issueInput, level: level, context: context)
    }


}
